code/model/ErrorPage.php

```diff
@@ -47,36 +69,32 @@ public function canAddChildren($member = null) {
 	 * file generated when the user hit's save and publish in the CMS
 	 *
 	 * @param int $statusCode
-	 *
 	 * @return SS_HTTPResponse
 	 */
 	public static function response_for($statusCode) {
 		// first attempt to dynamically generate the error page
-		$errorPage = ErrorPage::get()->filter(array(
-			"ErrorCode" => $statusCode
-		))->first(); 
+		$errorPage = ErrorPage::get()
+			->filter(array(
+				"ErrorCode" => $statusCode
+			))->first(); 
 
 		if($errorPage) {
 			Requirements::clear();
 			Requirements::clear_combined_files();
 
-			return ModelAsController::controller_for($errorPage)->handleRequest(
-				new SS_HTTPRequest('GET', ''), DataModel::inst()
-			);
+			return ModelAsController::controller_for($errorPage)
+				->handleRequest(
+					new SS_HTTPRequest('GET', ''),
+					DataModel::inst()
+				);
 		}
 		
 		// then fall back on a cached version
-		$cachedPath = self::get_filepath_for_errorcode(
```

Comment:
Yes, this is coupling, but we didn't put it in for jollies: we've tolerated the coupling in here because it implements an important feature. Refactor it out in a future PR, but don't just delete it in this one!

I don't think that making the state set API a blocker for 4.0-alpha1, which would be the alternative solution, is a good idea.

Please re-add the coupling to Translatable.